### PR TITLE
TXAD-26 Implemented Logbook List Screen

### PR DESCRIPTION
**Pull request by 404ajayreddy**

### CODE REVIEW

File: lib/blocs/LogBook/logbook_events.dart [hunk 1]
```diff
@@ -0,0 +1,7 @@
+abstract class LogbookEvents {}
```
Comment (ruchawajpe) [inline lib/blocs/LogBook/logbook_events.dart:1]: Remove abstract class. If you have more that one event use abstract class.

File: lib/screens/logBook/logbook_list_screen.dart [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  static List logbookData = [];
+
+  const LogbookListScreen({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    context.read<LogbookBloc>().add(FetchLogbookList(pageNo: 1));
+    return Scaffold(
+      appBar: const GenericAppBar(
+        title: 'Report a Log',
```
Comment (ruchawajpe) [inline lib/screens/logBook/logbook_list_screen.dart:22]: Check if we get the key for it else use String constants.

**Review by ruchawajpe — CHANGES_REQUESTED** [review]

File: lib/blocs/LogBook/logbook_states.dart [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+class FetchingLogbookList extends LogbookStates {}
+
+class LogbookListFetched extends LogbookStates {
+  final FetchLogBookListModel fetchLogBookListModel;
+
+  const LogbookListFetched({required this.fetchLogBookListModel});
+}
+
+class LogbookFetchError extends LogbookStates {}
```
[the remark this answers is not in the document]
Comment (404ajayreddy) [reply lib/blocs/LogBook/logbook_states.dart:17]: Need it for the bloc emit

File: lib/screens/logBook/widgets/logbook_list.dart [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+                      ),
+                    ),
+                  );
+                },
+                separatorBuilder: (context, index) {
+                  return const SizedBox(height: tinierSpacing);
+                });
+          }
+        }
+        if (state is LogbookFetchError) {
```
[the remark this answers is not in the document]
Comment (404ajayreddy) [reply lib/screens/logBook/widgets/logbook_list.dart:142]: Removed

File: lib/blocs/LogBook/logbook_events.dart (hunk 1 above)
Comment (404ajayreddy) [reply lib/blocs/LogBook/logbook_events.dart:1]: Will have a post event here for add log screen So need it

File: lib/screens/logBook/logbook_list_screen.dart (hunk 2 above)
Comment (404ajayreddy) [reply lib/screens/logBook/logbook_list_screen.dart]: done

File: lib/blocs/LogBook/logbook_events.dart (hunk 1 above)
Comment (ruchawajpe) [reply lib/blocs/LogBook/logbook_events.dart:1]: Alright

File: lib/blocs/LogBook/logbook_states.dart (hunk 3 above)
[the remark this answers is not in the document]
Comment (ruchawajpe) [reply lib/blocs/LogBook/logbook_states.dart:17]: Alright!

**Review by ruchawajpe — APPROVED** [review]

**Pull request merged**